### PR TITLE
CLDR-17048 Run Segmenter.java for new Unicode versions

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-17048

Tool is run from Unicodetools. 

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: common/segments/root.xml [hunk 1]
```diff
@@ -28,10 +28,14 @@ For terms of use, see http://www.unicode.org/copyright.html
...
-				<variable id="$LinkingConsonant">[\p{Gujr}\p{sc=Telu}\p{sc=Mlym}\p{sc=Orya}\p{sc=Beng}\p{sc=Deva}&amp;\p{Indic_Syllabic_Category=Consonant}]</variable>
+				<!-- Note: ConjunctLinkingScripts is not used anymore, instead that list exists in the derivation of Indic_Conjunct_Break. -->
+				<!-- It is kept here so that the diff of the generated test cases compared to the Unicode 15.1 β is minimal. -->
+				<!-- TODO(egg): Consider removing in Unicode 16.0. -->
+				<variable id="$ConjunctLinkingScripts">[\p{Gujr}\p{sc=Telu}\p{sc=Mlym}\p{sc=Orya}\p{sc=Beng}\p{sc=Deva}]</variable>
+				<variable id="$ConjunctLinker">\p{Indic_Conjunct_Break=Linker}</variable>
+				<variable id="$LinkingConsonant">\p{Indic_Conjunct_Break=Consonant}</variable>
 				<variable id="$ExtPict">\p{Extended_Pictographic}</variable>
-				<variable id="$ExtCccZwj">[[$Extend-\p{ccc=0}] $ZWJ]</variable>
+				<variable id="$ExtCccZwj">[\p{Indic_Conjunct_Break=Linker}\p{Indic_Conjunct_Break=Extend}]</variable>
```
Comment (pedberg-icu) [inline common/segments/root.xml:38]: Hmm. Should $ExtCccZwj still include $ZWJ? If so, where does that get added? If not, maybe $ExtCccZwj should renamed...

**Comment by macchiati** [comment]
I think that is in InCB=Extend.

Because the UTC had taken in the customizations CLDR had (with a new
property), in this release we're following
https://unicode-org.github.io/unicode-reports/tr29/tr29.html exactly.

File: common/segments/root.xml (hunk 1 above)
Comment (eggrobin) [reply common/segments/root.xml:38]: Scripsit @macchiati:
I think [ZWJ] is in InCB=Extend.
Indeed, see https://github.com/unicode-org/unicodetools/blob/9595f090650e99e3e752b37a7a3866ac8a91999b/unicodetools/src/main/java/org/unicode/text/UCD/ToolUnicodePropertySource.java#L1193-L1203.

**Comment by macchiati** [comment]
Today is my last day for a while. Can someone review?

**Review by eggrobin — APPROVED** [review]

**Review by pedberg-icu — APPROVED** [review]

**Comment by pedberg-icu** [comment]
@macchiati 
> Today is my last day for a while. Can someone review?

It is approved...

**Comment by macchiati** [comment]
Thanks all!

**Pull request merged**